### PR TITLE
Enable/disable media file delete confirmation

### PR DESCRIPTION
**Pull request by bnk1**
Disable/enable delete confirmation.

## Description

Disable/enable delete confirmation option.

## Motivation and context

On cases where you need to delete multiple files, it can be annoying.

## How has this been tested?

On my machine.

## What is the effect on users?

None.
## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
Thanks for this contribution. 

Despite from what you cover, there are other delete operations possible in Kodi. Just to give an example, definitely not deleting of PVR recordings. In my opinion we should introduce a global setting that covers all delete operations that can be done in Kodi. This PR would be a start. Or you name the setting exactly after what it covers exactly.

**Comment by ksooo** [comment]
It seems you want to tackle videos, but at least the change in `CGUIMediaWindow` could also cover other media, like music, as this is a base class used for different media types.

**Comment by bnk1** [comment]
No problem. I can make it for just videos/photos/Music or for everything. I tried to cover everything but at first it was just videos.

**Comment by ksooo** [comment]
> Let me know what is missing and I'll add it.

Well, that's a bit of a problem, at least for me. You need to find out yourself the functionality, then change the respective code. 

What I can say that PVR (Recordings) and Favourites also offers functionality to delete items. But there might be even more.

File: system/settings/settings.xml [hunk 1]
```diff
@@ -1075,6 +1075,19 @@
           </dependencies>
           <control type="toggle" />
         </setting>
+	<setting id="filelists.deleteconfirm" type="boolean" label="40801" help="40802">
```
Comment (ksooo) [inline system/settings/settings.xml:1078]: I'm not sure this setting belongs here. Is this handling always confirmation when deleting physical files?

**Review by ksooo — COMMENTED** [review]
I'm not sure about the actual use cases the code changes cover. Could you please name them in this PR?

File: xbmc/video/dialogs/GUIDialogVideoInfo.cpp [hunk 2]
```diff
@@ -1483,8 +1483,12 @@ bool CGUIDialogVideoInfo::DeleteVideoItem(const std::shared_ptr<CFileItem>& item
         item->m_bIsFolder = true;
 
       CGUIComponent *gui = CServiceBroker::GetGUI();
-      if (gui && gui->ConfirmDelete(item->GetPath()))
-        CFileUtils::DeleteItem(item);
+
+        if (gui) // BOAZ_N
```
[the remark this answers is not in the document]
Comment (bnk1) [reply xbmc/video/dialogs/GUIDialogVideoInfo.cpp:1487]: Of course, it was marked for personal uses on my fork.

File: xbmc/video/dialogs/GUIDialogVideoInfo.cpp [hunk 3]
```diff
@@ -1483,8 +1483,12 @@ bool CGUIDialogVideoInfo::DeleteVideoItem(const std::shared_ptr<CFileItem>& item
...
+        {
+            if (!CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_FILELISTS_DELETECONFIRM) || gui->ConfirmDelete(item->GetPath()))
+                CFileUtils::DeleteItem(item);
```
[the remark this answers is not in the document]
Comment (bnk1) [reply xbmc/video/dialogs/GUIDialogVideoInfo.cpp:1490]: Yep. But at first I only wanted it for videos.

**Comment by bnk1** [comment]
> > Let me know what is missing and I'll add it.
> 
> Well, that's a bit of a problem, at least for me. You need to find out yourself the functionality, then change the respective code.
> 
> What I can say that PVR (Recordings) and Favourites also offers functionality to delete items. But there might be even more.
> I'm not sure about the actual use cases the code changes cover. Could you please name them in this PR?

I think it is better to make the option(s) only for videos and pictures.

**Comment by bnk1** [comment]
> I've made some formatting changes to meet the current code style. The diffs are available in the following links:
> 
> * [PR25857-0001-c2def83042-Disable-enable-delete-confirmation.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31675/artifact/PR25857-0001-c2def83042-Disable-enable-delete-confirmation.diff)
> * [PR25857-0002-541304f0ff-Changed-to-video-delete-confirmati.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31675/artifact/PR25857-0002-541304f0ff-Changed-to-video-delete-confirmati.diff)
> * [PR25857.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31675/artifact/PR25857.diff)
> 
> For more information please see our [current code style guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

Sure, no problem.

**Comment by bnk1** [comment]
All done (just videos).
We can implement it for all files on the confirmDelete function as well (separately).

**Comment by bnk1** [comment]
> Thanks for this contribution.
> 
> Despite from what you cover, there are other delete operations possible in Kodi. Just to give an example, definitely not deleting of PVR recordings. In my opinion we should introduce a global setting that covers all delete operations that can be done in Kodi. This PR would be a start. Or you name the setting exactly after what it covers exactly.

This can be done on the confirmation dialog box function.

**Comment by bnk1** [comment]
> I've made some formatting changes to meet the current code style. The diffs are available in the following links:
> 
> * [PR25857-0001-c2def83042-Disable-enable-delete-confirmation.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31676/artifact/PR25857-0001-c2def83042-Disable-enable-delete-confirmation.diff)
> * [PR25857-0002-541304f0ff-Changed-to-video-delete-confirmati.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31676/artifact/PR25857-0002-541304f0ff-Changed-to-video-delete-confirmati.diff)
> * [PR25857.diff](https://jenkins.kodi.tv/job/BuildMulti-PR/31676/artifact/PR25857.diff)
> 
> For more information please see our [current code style guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

It is already updated accordingly (?)
How do I add the MILESTONE and LABEL?

**Comment by bnk1** [comment]
Changed to "Confirm file delete" for all files where the confirmation dialog may appear.

**Comment by ksooo** [comment]
> It is already updated accordingly (?)

 Please sqush down your changes to a single commit. 
 
> How do I add the MILESTONE and LABEL?

 Will be done by team Kodi.

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 4]
```diff
@@ -1060,8 +1060,15 @@ void CGUIWindowVideoBase::OnDeleteItem(const CFileItemPtr& item)
...
-      CFileUtils::DeleteItem(item);
+    
+    if (gui)
+    {
+      bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+        CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
+      
+      if (!confirm || gui->ConfirmDelete(item->GetPath()))
+        CFileUtils::DeleteItem(item);
+    }
```
[the remark this answers is not in the document]
Comment (bnk1) [reply xbmc/video/windows/GUIWindowVideoBase.cpp:1071]: You are correct but the CUtil/CFileUtils can't create a gui object.
On the other hand, the ConfirmDelete function doesn't have an access to the flag.

**Comment by bnk1** [comment]
Done.

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (bnk1) [reply xbmc/video/windows/GUIWindowVideoBase.cpp]: Done.

**Comment by bnk1** [comment]
Is there a quick way to squash the commits of the fork?

File: xbmc/test/TestUtil.cpp [hunk 5]
```diff
@@ -6,6 +6,7 @@
  *  See LICENSES/README.md for more information.
  */
 
+#include "FileItem.h"
```
Comment (ksooo) [inline xbmc/test/TestUtil.cpp:9]: Please remove. Not needed.

File: xbmc/Util.h [hunk 6]
```diff
@@ -227,6 +227,12 @@ class CUtil
                                    const std::string& file,
                                    const std::string& destname = "");
 
+  /*!
+   * \brief Delete a file with/without confirmation.
+   *        Depends on confirmation flag in settings.
+   */
+  static bool FileDelWithConfirm(CFileItemPtr pItem);
```
Comment (ksooo) [inline xbmc/Util.h:234]: 1) We are not short on letters. Please change function name to `FileDeleteWithConfirm`
2) Change param type to `const CFileItem&`, please. No need for a (smart) pointer here. Reference is fine.

File: xbmc/test/TestUtils.cpp [hunk 7]
```diff
@@ -7,6 +7,8 @@
  */
 
 #include "TestUtils.h"
+
+#include "FileItem.h"
```
Comment (ksooo) [inline xbmc/test/TestUtils.cpp:11]: Please remove. Not needed.

File: xbmc/Util.cpp [hunk 8]
```diff
@@ -2467,3 +2468,19 @@ void CUtil::CopyUserDataIfNeeded(const std::string& strPath,
...
+  if (gui)
+  {
+    const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+        CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
+
+    if (!confirm || gui->ConfirmDelete(pItem->GetPath()))
+      return CFileUtils::DeleteItem(pItem);
+  }
+
+  return false;
```
Comment (ksooo) [inline xbmc/Util.cpp:2485]: No need to check for `gui` presence first. Settings manager works also without GUI. Should read, like:

```c++
bool deleteItem = false;

if (CServiceBroker::GetSettingsComponent()->GetS...
{
  CGUIComponent* gui = CService...
  if (gui)
    deleteItem = gui->Confirm...
}
else
{
  deleteItem = true;
}

if (deleteItem)
 return CFileUtil::DeleteItem...
 
return false;
```

**Review by ksooo — COMMENTED** [review]
Needs a bit more love still.

File: addons/resource.language.en_gb/resources/strings.po [hunk 9]
```diff
@@ -24700,3 +24700,14 @@ msgid "50th Anniversary Edition"
 msgstr ""
 
 # 40000 to 40800 are reserved for Video Versions feature
+
+#: system/settings/settings.xml
+msgctxt "#40801"
+msgid "Confirm file deletion"
```
Comment (CastagnaIT) [inline addons/resource.language.en_gb/resources/strings.po:24706]: this must be a question, not a statement

File: addons/resource.language.en_gb/resources/strings.po [hunk 10]
```diff
@@ -24700,3 +24700,14 @@ msgid "50th Anniversary Edition"
...
+msgstr ""
+
+#. Description of setting with label #40801 "Confirm file deletion"
+#: system/settings/settings.xml
+msgctxt "#40802"
+msgid "Confirm file deletion"
+msgstr ""
```
Comment (CastagnaIT) [inline addons/resource.language.en_gb/resources/strings.po:24713]: missing file endline

File: addons/resource.language.en_gb/resources/strings.po (hunk 9 above)
Comment (CastagnaIT) [inline addons/resource.language.en_gb/resources/strings.po:24706]: This not only asks to delete files but you involved it in to profile management
to me its a bit inconsistent

File: system/settings/settings.xml [hunk 11]
```diff
@@ -1075,6 +1075,19 @@
...
+			  <dependencies>
+				  <dependency type="enable">
+					  <or>
+						  <condition on="property" operator="!is" name="ProfileHasFilesLocked" />
+						  <condition on="property" name="IsMasterUser" />
+					  </or>
+				  </dependency>
+			  </dependencies>
+			  <control type="toggle" />
+		  </setting>
```
Comment (CastagnaIT) [inline system/settings/settings.xml:1090]: you have to use spaces instead of tab's

**Comment by ksooo** [comment]
> This not only asks to delete files but you involved it in to profile management

Exactly the "problem" I still have with this PR - I have no good understanding for which use cases the code change in this PR has effect, whether there is something missing (could be added in a follow-up PR), but more important whether the setting now will be used for something that is not deleting files.

After reading the setting description, users could expect that whenever I delete a file in Kodi, this setting will be taken into account. Is this actually the case with the current implementation as is in this PR?

For example:
1) when I delete files using Kodi file manager, will the setting being taken into account?It should.
2) when I remove an item from video library, will the setting being taken into account? It should, but only when the actual video file shall be deleted, not for removal of only the the entry in video db
3) when I delete a PVR recording, will this setting being taken into account

Don't get me wrong, I'm not that much concerned to have forgotten something, it is more that this setting will be applied to some non-file delete activities...

**Comment by CastagnaIT** [comment]
as "Confirm file deletion" theoretically i expect that refer to real local files only, not database,
but now that you point this out i think it could be misunderstood by users...
should be clarified better in the setting description the difference?

if you want keep the new setting in to "Media" setting,
i would avoid involving components different than "library" such as the profile management
also in the future an oversight by some other dev could add the use of this setting inappropriately elsewhere...
you should at least add a comment in the code to use this setting only for library tasks

if instead the final goal is to apply this globally on kodi,
it would make more sense add the new setting entry in the "System" settings, not "Media"
so it's clearer that it's globally applied

**Comment by bnk1** [comment]
You are correct. We should implement it for Videos and Pictures only though.

**Comment by bnk1** [comment]
> > This not only asks to delete files but you involved it in to profile management
> 
> Exactly the "problem" I still have with this PR - I have no good understanding for which use cases the code change in this PR has effect, whether there is something missing (could be added in a follow-up PR), but more important whether the setting now will be used for something that is not deleting files.
> 
> After reading the setting description, users could expect that whenever I delete a file in Kodi, this setting will be taken into account, no matter where in Kodi I do it. Is this actually the case with the current implementation as is in this PR?
> 
> For example:
> 
> 1. when I delete files using Kodi file manager, will the setting being taken into account?It should.
> 2. when I remove an item from video library, will the setting being taken into account? It should, but only when the actual video file shall be deleted, not for removal of only the the entry in video db
> 3. when I delete a PVR recording, will this setting being taken into account? It should.
> 
> Don't get me wrong, I'm not that much concerned to have forgotten something, it is more that this setting could be applied to some non-file delete activities...

Correct. That's why at first I implemented it only for videos (it should be added for pictures as well).

File: xbmc/test/TestUtils.cpp (hunk 7 above)
Comment (bnk1) [reply xbmc/test/TestUtils.cpp:11]: It doesn't compile without.

File: xbmc/Util.h (hunk 6 above)
Comment (bnk1) [reply xbmc/Util.h:234]: 1. Can't be implemented in file utils as there is no access to the settings/flags there.
2-4. Ok

**Comment by bnk1** [comment]
Isn't there a way to use the clang formatting to affect spaces/tabs coding conventions?

File: xbmc/Util.h (hunk 6 above)
Comment (ksooo) [reply xbmc/Util.h:234]: > Can't be implemented in file utils as there is no access to the settings/flags there.

why can’t you just include the missing headers?

File: xbmc/test/TestUtils.cpp (hunk 7 above)
Comment (ksooo) [reply xbmc/test/TestUtils.cpp:11]: Compiler error output, please.

Comment (bnk1) [reply xbmc/test/TestUtils.cpp:11]: Severity	Code	Description	Project	File	Line	Suppression State
Error	C2143	syntax error: missing ',' before '&' (compiling source file C:\Code\xbmc\xbmc\test\TestUtils.cpp)	kodi-test	C:\Code\xbmc\xbmc\Util.h	232

File: xbmc/Util.h (hunk 6 above)
Comment (bnk1) [reply xbmc/Util.h:234]: It's not missing headers: "error C2027: use of undefined type 'CGUIComponent'"

Comment (ksooo) [reply xbmc/Util.h]: That's exactly telling that there is an include missing! You need to add

```
#include "guilib/GUIComponent.h"
```

in CFileUtils.h

File: xbmc/test/TestUtils.cpp (hunk 7 above)
Comment (ksooo) [reply xbmc/test/TestUtils.cpp:11]: This is not caused by "CFilUtils.cpp". Nees to be fixed in "Util.h". There, `#include "FileItem.h` is missing. You added there a method having a `CFileItemPtr` as parameter. For that, FileItem.h needs to be included.

File: addons/resource.language.en_gb/resources/strings.po [hunk 12]
```diff
@@ -24712,3 +24712,14 @@ msgid "50th Anniversary Edition"
 msgstr ""
 
 # 40000 to 40800 are reserved for Video Versions feature
+
+#: system/settings/settings.xml
```
Comment (ksooo) [inline addons/resource.language.en_gb/resources/strings.po:24716]: Please add `#.` line, like you did for the other string you've introduced, explaining in clear words what this setting is actually controlling. Our translators need this.

File: system/settings/settings.xml [hunk 13]
```diff
@@ -1080,6 +1080,19 @@
...
+        <setting id="filelists.confirmfiledelete" type="boolean" label="40801" help="40802">
+          <level>2</level>
+          <default>false</default>
+          <dependencies>
+            <dependency type="enable">
+              <or>
+                <condition on="property" operator="!is" name="ProfileHasFilesLocked" />
+                <condition on="property" name="IsMasterUser" />
+              </or>
+            </dependency>
```
Comment (ksooo) [inline system/settings/settings.xml:1092]: Can you explain why we need this dependency? Just asking.

File: xbmc/Util.h [hunk 14]
```diff
@@ -227,6 +227,10 @@ class CUtil
                                    const std::string& file,
                                    const std::string& destname = "");
 
+  /** \brief Delete a file with confirmation unless setting is off.
+  */
+  static bool FileDeleteWithConfirm(const CFileItemPtr& item);
```
Comment (ksooo) [inline xbmc/Util.h:232]: One more time: 

1) This function should be moved to `CFileUtils`.
2) Replace `CFileItemPtr` with `std::shared_ptr>CFileItem>`.

File: xbmc/Util.cpp [hunk 15]
```diff
@@ -2467,3 +2468,19 @@ void CUtil::CopyUserDataIfNeeded(const std::string& strPath,
     CFile::Copy(srcPath, destPath);
   }
 }
+
+bool CUtil::FileDeleteWithConfirm(const CFileItemPtr& item)
```
Comment (ksooo) [inline xbmc/Util.cpp:2472]: Replace `CFileItemPtr` with `std::shared_ptr>CFileItem>`.

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/Util.cpp (hunk 15 above)
Comment (enen92) [reply xbmc/Util.cpp:2472]: Didn´t we have a GUI utils or something similar @ksooo? I would avoid adding this to this generic utils class

Comment (ksooo) [reply xbmc/Util.cpp:2472]: We have `CFileUtils` which already has GUI deps and I already proposed to move this code there.

Comment (ksooo) [reply xbmc/Util.cpp:2472]:
```c++
bool CFileUtils::RenameFile(const std::string &strFile)
{
  std::string strFileAndPath(strFile);
  ...
  if (CGUIKeyboardFactory::ShowAndGetInput(strFileName, CVariant{g_localizeStrings.Get(16013)}, false))
```

Comment (ksooo) [reply xbmc/Util.cpp:2472]: So, having `CFileUtils::DeleteFile` would fit quite nicely.

Comment (bnk1) [reply xbmc/Util.cpp:2472]: That's what I tried at first but it didn't compile.

**Comment by bnk1** [comment]
> @bnk1 while trying to squash (or whatever) you now messed up the commit of this PR.
> 
> What you could do to rebase to latest upstream master and squash to a single commit is:
> 
> ```
> git remote add upstream https://github.com/xbmc/xbmc.git
> git fetch upstream
> git rebase upstream/master
> git reset upstream/master
> git commit -a -m"whatever you want to put as commit message"
> git push orgin -ff master
> ```

I know, but this didn't work. 
Maybe open another fork for the commit/push?

File: system/settings/settings.xml (hunk 13 above)
Comment (bnk1) [reply system/settings/settings.xml:1092]: Same as the one on Allow file deletion. Revise if needed.

File: addons/resource.language.en_gb/resources/strings.po (hunk 12 above)
Comment (bnk1) [reply addons/resource.language.en_gb/resources/strings.po:24715]: Done

File: xbmc/Util.h (hunk 14 above)
Comment (bnk1) [reply xbmc/Util.h]: Done

**Comment by bnk1** [comment]
> @bnk1 while trying to squash (or whatever) you now messed up the commit of this PR.
> 
> What you could do to rebase to latest upstream master and squash to a single commit is:
> 
> ```
> git remote add upstream https://github.com/xbmc/xbmc.git
> git fetch upstream
> git rebase upstream/master
> git reset upstream/master
> git commit -a -m"whatever you want to put as commit message"
> git push orgin -ff master
> ```

Now it worked.
Thanks!

**Comment by ksooo** [comment]
> Now it worked.

Still seeing 23 commit here?!

File: addons/resource.language.en_gb/resources/strings.po (hunk 12 above)
Comment (ksooo) [reply addons/resource.language.en_gb/resources/strings.po:24715]: Well, is it helpful just to repeat the string? Not really. Please look how others did it, explaining where and for what exactly it is used, like "label for a setting to configure whether video files ..."

File: system/settings/settings.xml (hunk 13 above)
Comment (ksooo) [reply system/settings/settings.xml:1092]: Then this rather should be a dependency to "allow file deletion". Enable this setting only if "allow file deletion" is enabled.

File: xbmc/Util.cpp [hunk 16]
```diff
@@ -72,6 +72,7 @@
 #include "settings/SettingsComponent.h"
 #include "utils/Digest.h"
 #include "utils/FileExtensionProvider.h"
+#include "utils/FileUtils.h"
```
Comment (ksooo) [inline xbmc/Util.cpp:75]: Remove this, not needed.

File: xbmc/profiles/ProfileManager.cpp [hunk 17]
```diff
@@ -513,7 +513,8 @@ bool CProfileManager::DeleteProfile(unsigned int index)
   item->m_bIsFolder = true;
   item->Select(true);
 
-  CGUIComponent *gui = CServiceBroker::GetGUI();
+  CGUIComponent* gui = CServiceBroker::GetGUI();
+
```
Comment (ksooo) [inline xbmc/profiles/ProfileManager.cpp:517]: Revert this changes, please. This is only leftover from a previous iteration of your PR. You do not want to change profile manager.

File: xbmc/test/TestUtils.cpp [hunk 18]
```diff
@@ -7,6 +7,7 @@
  */
 
 #include "TestUtils.h"
+
```
Comment (ksooo) [inline xbmc/test/TestUtils.cpp:10]: Remove this empty line. This is only a leftover from a previous iteration of the PR. You do not want to change this file.

File: xbmc/utils/FileUtils.cpp [hunk 19]
```diff
@@ -7,7 +7,6 @@
  */
 
 #include "FileUtils.h"
-
```
Comment (ksooo) [inline xbmc/utils/FileUtils.cpp:10]: Revert the removal of this line. We always keep an empty line between header include for the class implemented in the cpp file.

File: xbmc/utils/FileUtils.cpp [hunk 20]
```diff
@@ -353,3 +353,20 @@ bool CFileUtils::Exists(const std::string& strFileName, bool bUseCache)
...
   return CFile::Exists(strFileName, bUseCache);
 }
+
+
+bool CFileUtils::DeleteWithConfirm(const std::shared_ptr<CFileItem>& item)
+{
+  const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+      CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
+
+  if (confirm)
```
Comment (ksooo) [inline xbmc/utils/FileUtils.cpp:363]: You could save the `confirm`variable and just write

```
if (CServiceBroker::GetSettingsComponent...)
```

File: xbmc/utils/FileUtils.h [hunk 21]
```diff
@@ -10,6 +10,7 @@
 
 #include <memory>
 #include <string>
+#include "FileItem.h"
```
Comment (ksooo) [inline xbmc/utils/FileUtils.h:13]: No need for this include. remove it.
Instead, add `#include <memory>`- for `std::shared_ptr`

File: xbmc/utils/FileUtils.cpp (hunk 20 above)
Comment (bnk1) [reply xbmc/utils/FileUtils.cpp:363]: I know. I think it is more readable this way.

**Comment by bnk1** [comment]
> Still seeing 23 commit here?!

What do you mean?

**Comment by bnk1** [comment]
> > @bnk1 while trying to squash (or whatever) you now messed up the commit of this PR.
> > What you could do to rebase to latest upstream master and squash to a single commit is:
> > ```
> > git remote add upstream https://github.com/xbmc/xbmc.git
> > git fetch upstream
> > git rebase upstream/master
> > git reset upstream/master
> > git commit -a -m"whatever you want to put as commit message"
> > git push orgin -ff master
> > ```

After 'git rebase upstream/master' it keeps making conflicts rather than just take the last version (?)

**Comment by bnk1** [comment]
> git push orgin -ff master

1. Keep solving the conflicts using HEAD.
2. git push origin -ff master

Now it is squashed (?).

**Comment by bnk1** [comment]
> > Now it worked.
> 
> Still seeing 23 commit here?!

Yeah... :)
Now only one...

File: xbmc/utils/FileUtils.h [hunk 22]
```diff
@@ -21,6 +21,10 @@ class CFileUtils
   static bool DeleteItem(const std::shared_ptr<CFileItem>& item);
   static bool DeleteItem(const std::string &strPath);
   static bool Exists(const std::string& strFileName, bool bUseCache = true);
+  
+  /** \brief Delete a file with confirmation unless setting is off.
+  */
```
Comment (CastagnaIT) [inline xbmc/utils/FileUtils.h:26]: bad formatting style,
and since this must used only for audio/video/music you should specify it,
because this class is generic
```c++
/*
 * \brief Deleting a file with confirmation according to settings, this applies only to audio/video/music files.
 */
```

File: xbmc/utils/FileUtils.cpp [hunk 23]
```diff
@@ -353,3 +354,19 @@ bool CFileUtils::Exists(const std::string& strFileName, bool bUseCache)
 {
   return CFile::Exists(strFileName, bUseCache);
 }
+
+bool CFileUtils::DeleteWithConfirm(const std::shared_ptr<CFileItem>& item)
+{
+  const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+      CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
```
Comment (CastagnaIT) [inline xbmc/utils/FileUtils.cpp:361]: @ksooo what worring me is that DeleteWithConfirm use CFileUtils that is generic, can be ok but
this setting should not be used for files that are not library related (audio/video/music) but nowhere is it specified, and no file type check happens here

File: xbmc/utils/FileUtils.h (hunk 22 above)
Comment (ksooo) [reply xbmc/utils/FileUtils.h:26]: Agree to formatting, not to charging the description. The implementation deletes whatever it gets. The fact for what is currently used is nothing for the description of the function. Keep as is, please.

File: xbmc/utils/FileUtils.cpp (hunk 23 above)
Comment (ksooo) [reply xbmc/utils/FileUtils.cpp:361]: See my other comment. Don’t mix up usage and implantation. No checks for files belong here. This is controlled from outside, by the caller.

Comment (CastagnaIT) [reply xbmc/utils/FileUtils.cpp:361]: if so, this code
```

  const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
      CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
```

is really wrong.
the `filelists.confirmfiledelete` is specified for audio/video/music and here IS NOT SO
then or you move this setting OUT of the "Media" to "System" and you made it generic for all kodi sections
or as is this is implemented is no good

Comment (ksooo) [reply xbmc/utils/FileUtils.cpp:361]: I agree, the setting belongs into another section of the settings, not system. At least for now.

Comment (bnk1) [reply xbmc/utils/FileUtils.cpp:361]: Actually this setting is meant to refer to the previous setting (allow file deletion). If the implementation of a generic method isn’t compatible with it, we should remove the generic method and implement it specifically into the locations where it is used. 
what cane be done is use the generic function and pass it the setting.
Personally, I don’t like the long line code of getting the setting, it’s not readable.
I don’t think this should be a generic for all files.  Everytime I make a new Kodi installation I change this setting and this feature was meant for that.

Comment (bnk1) [reply xbmc/utils/FileUtils.cpp]: Solved

**Comment by ksooo** [comment]
The latest changes go into the wrong direction, I’m afraid.

File: xbmc/utils/FileUtils.cpp (hunk 23 above)
Comment (ksooo) [reply xbmc/utils/FileUtils.cpp]: > what cane be done is use the generic function and pass it the setting.

NO! This is imo not the way to go.

File: system/settings/settings.xml (hunk 13 above)
Comment (ksooo) [inline system/settings/settings.xml:1092]: Change to:

```suggestion
            <dependency type="enable">
              <condition setting="filelists.allowfiledeletion" operator="is">true</condition>
            </dependency>
```

File: xbmc/test/TestUtils.cpp (hunk 18 above)
Comment (ksooo) [inline xbmc/test/TestUtils.cpp:10]: One more time: Remove this empty line. This is only a leftover from a previous iteration of the PR. You do not want to change this file.

File: xbmc/utils/FileUtils.cpp [hunk 24]
```diff
@@ -353,3 +354,19 @@ bool CFileUtils::Exists(const std::string& strFileName, bool bUseCache)
 {
   return CFile::Exists(strFileName, bUseCache);
 }
+
+bool CFileUtils::DeleteItemWithConfirm(const std::shared_ptr<CFileItem>& item)
+{
+  bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
```
Comment (ksooo) [inline xbmc/utils/FileUtils.cpp:360]:
```suggestion
 const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
```

File: xbmc/video/dialogs/GUIDialogVideoInfo.cpp [hunk 25]
```diff
@@ -1482,9 +1482,7 @@ bool CGUIDialogVideoInfo::DeleteVideoItem(const std::shared_ptr<CFileItem>& item
       if (item->IsStack())
         item->m_bIsFolder = true;
 
-      CGUIComponent *gui = CServiceBroker::GetGUI();
-      if (gui && gui->ConfirmDelete(item->GetPath()))
-        CFileUtils::DeleteItem(item);
+        CFileUtils::DeleteItemWithConfirm(item);
```
Comment (ksooo) [inline xbmc/video/dialogs/GUIDialogVideoInfo.cpp:1485]: Wrong indentation. Jenkins already told you.

File: xbmc/video/windows/GUIWindowVideoNav.cpp [hunk 26]
```diff
@@ -712,9 +712,8 @@ void CGUIWindowVideoNav::OnDeleteItem(const CFileItemPtr& pItem)
            m_vecItems->IsPath("special://videoplaylists/"))
   {
     pItem->m_bIsFolder = false;
-    CGUIComponent *gui = CServiceBroker::GetGUI();
-    if (gui && gui->ConfirmDelete(pItem->GetPath()))
-      CFileUtils::DeleteItem(pItem);
+
```
Comment (ksooo) [inline xbmc/video/windows/GUIWindowVideoNav.cpp:715]: Remove this blank line.

**Review by ksooo — CHANGES_REQUESTED** [review]
We are close...

File: system/settings/settings.xml [hunk 27]
```diff
@@ -1080,6 +1080,16 @@
           </dependencies>
           <control type="toggle" />
         </setting>
+        <setting id="filelists.confirmfiledelete" type="boolean" label="40801" help="40802">
+          <level>2</level>
+          <default>false</default>
```
Comment (ksooo) [inline system/settings/settings.xml:1085]: Oh, this should definitely default to "True"! Which means to confirm deletion.

```suggestion
          <default>true</default>
```

File: xbmc/music/windows/GUIWindowMusicNav.cpp [hunk 28]
```diff
@@ -826,9 +826,8 @@ bool CGUIWindowMusicNav::OnContextButton(int itemNumber, CONTEXT_BUTTON button)
     if (PLAYLIST::IsPlayList(*item) || PLAYLIST::IsSmartPlayList(*item))
     {
       item->m_bIsFolder = false;
-      CGUIComponent *gui = CServiceBroker::GetGUI();
-      if (gui && gui->ConfirmDelete(item->GetPath()))
-        CFileUtils::DeleteItem(item);
+
```
Comment (ksooo) [inline xbmc/music/windows/GUIWindowMusicNav.cpp:829]: Remove this empty line, please.

File: xbmc/settings/Settings.h [hunk 29]
```diff
@@ -56,6 +56,7 @@ class CSettings : public CSettingsBase, public CSettingCreator, public CSettingC
   static constexpr auto SETTING_FILELISTS_SHOWEXTENSIONS = "filelists.showextensions";
   static constexpr auto SETTING_FILELISTS_IGNORETHEWHENSORTING = "filelists.ignorethewhensorting";
   static constexpr auto SETTING_FILELISTS_ALLOWFILEDELETION = "filelists.allowfiledeletion";
+  static constexpr auto SETTING_FILELISTS_CONFIRMFILEDELETE = "filelists.confirmfiledelete";
```
Comment (ksooo) [inline xbmc/settings/Settings.h:59]: To be consistant with naming, we should name this setting after what we have for allow file deletion!

```suggestion
  static constexpr auto SETTING_FILELISTS_CONFIRMFILEDELETION = "filelists.confirmfiledeletion";
 ```

**Review by ksooo — CHANGES_REQUESTED** [review]
The settings value default has to be "true", which means to confirm.

**Review by ksooo — COMMENTED** [review]
We are almost there. Please change the string as requested and squash down to a single commit, then we are ready to merge imo.

**Comment by ksooo** [comment]
And now please squash down to a single commit, then we are ready to merge your PR

**Review by ksooo — APPROVED** [review]
Thank you very much for your contribution and for your patience. It was fun working with you. Anytime again.

**Comment by ksooo** [comment]
jenkinks build this please

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Android build errors are unrelated.

**Comment by bnk1** [comment]
> Thank you very much for your contribution and for your patience. It was fun working with you. Anytime again.

Thank you very much for your help, it was a pleasure working with you too :)
